packages/errors/src/errors/assertion.test.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { assertError, isError } from './assertion';
+import { NotFoundError } from './common';
+import { CustomErrorBase } from './CustomErrorBase';
+
+const areErrors = [
+  { name: 'e', message: '' },
+  new Error(),
+  new NotFoundError(),
+  Object.create({ name: 'e', message: '' }),
+  Object.assign(Object.create({ name: 'e' }), {
+    get message() {
+      return '';
+    },
+  }),
+  new (class extends class {
+    message = '';
+  } {
+    name = 'e';
+  })(),
+  new (class extends CustomErrorBase {})(),
+  new (class extends NotFoundError {})(),
+];
+
+const notErrors = [
+  null,
+  0,
+  'loller',
+  Symbol(),
+  [],
+  BigInt(0),
+  false,
+  true,
+  { name: 'e' },
+  { message: '' },
+  { name: '', message: 'oh no' },
```

Comment:
Oh because empty name

Reply:
Makes me wonder if it makes sense to have a `stringifyError` though, that calls toString and can check for `[object Object]` and pick `.message` instead then etc. It could even gracefully handle thrown non-`ErrorLike` things sensibly